### PR TITLE
OCC-302: Fix Serializing the Region type using System.Text.Json throws an exception

### LINKED ISSUE #503: Serializing the Region type using System.Text.Json throws an exception. (OCC-302)
### Describe the bug

It throws a `System.NotSupportedException` with the following message:
Deserialization of types without a parameterless constructor, a singular parameterized constructor, or a parameterized constructor annotated with `JsonConstructorAttribute` is not supported. Type `OrchardCore.Commerce.AddressDataType.Region`.

### To Reproduce

Easiest way to reproduce is:

```csharp
using OrchardCore.Commerce.AddressDataType;
using System.Text.Json;

var json = JsonSerializer.Serialize(Regions.All);
var regions = JsonSerializer.Deserialize<IEnumerable<Region>>(json); // throws
```

### Expected behavior

Should not throw.

[Jira issue](https://lombiq.atlassian.net/browse/OCC-302)

### PR DESCRIPTION
**Pull request by sarahelsaig**
[OCC-302](https://lombiq.atlassian.net/browse/OCC-302)
Fixes #503

### CODE REVIEW

File: src/Libraries/OrchardCore.Commerce.AddressDataType/Region.cs [hunk 1]
```diff
@@ -4,7 +4,9 @@ namespace OrchardCore.Commerce.AddressDataType;
 
 public record Region(string EnglishName, string TwoLetterISORegionName, string DisplayName)
 {
-    public Region(RegionInfo info)
-        : this(info.EnglishName, info.TwoLetterISORegionName, info.EnglishName)
-    { }
+    // This used to be a constructor, but has been turned into a factory method to avoid causing JSON serialization
+    // exceptions. Seehttps://github.com/dotnet/runtime/issues/45373#issuecomment-812091894 for more info on the
```
Comment (wAsnk) [inline src/Libraries/OrchardCore.Commerce.AddressDataType/Region.cs:8]:
```suggestion
    // exceptions. See https://github.com/dotnet/runtime/issues/45373#issuecomment-812091894 for more info on the
```

File: src/Modules/OrchardCore.Commerce/Models/PromotionAndTaxProviderContext.cs [hunk 2]
```diff
@@ -22,13 +22,16 @@ public record PromotionAndTaxProviderContext(
     bool Stored = false,
     bool IsCorporation = false)
 {
-    public PromotionAndTaxProviderContext(
+    // This used to be a constructor, but has been turned into a factory method to avoid causing JSON serialization
+    // exceptions. Seehttps://github.com/dotnet/runtime/issues/45373#issuecomment-812091894 for more info on the
```
Comment (wAsnk) [inline src/Modules/OrchardCore.Commerce/Models/PromotionAndTaxProviderContext.cs:26]:
```suggestion
    // exceptions. See https://github.com/dotnet/runtime/issues/45373#issuecomment-812091894 for more info on the
```

**Comment by wAsnk** [comment]
I'm not sure what's going on with the tests but it seems like there is a failing one without failing the tests?
OrderEditorShouldWorkProperly shows up as failed in the summary:
image

But not here:
image

**Review by wAsnk — CHANGES_REQUESTED** [review]

**Comment by sarahelsaig** [comment]
> I'm not sure what's going on with the tests but it seems like there is a failing one without failing the tests?

The tests were awaited, but did not throw on failure. Fixed now.

**Review by wAsnk — APPROVED** [review]

**Pull request merged**